[DOM Reference] Update Event interface reference

teoli2003 opened the pull request:
This is part of #9740.

- Interface and its children are all documented, and in the modern format.
- All are listed.
- Non-standard msConvertURL old-IE only deleted.
- I kept DOMHighResTimestamp, we will fix these all together.
- Tags have been fixed.

Not done: guide stayed there (part of phase 2)

teoli2003 committed:
Update Event interface reference

teoli2003 committed:
Fix typo

teoli2003 committed:
Fix Note.

teoli2003 committed:
Fix missing front-matter for bcd

teoli2003 committed:
Fix Event.target

wbamberg reviewed (changes requested):
Thanks @teoli2003 , nice work!

teoli2003 committed:
Improve grammar

Co-authored-by: wbamberg <will@bootbonnet.ca>

teoli2003 committed:
Use deprecated rather than the complex legacy alias.

Co-authored-by: wbamberg <will@bootbonnet.ca>

teoli2003 committed:
Simplify grammar

Co-authored-by: wbamberg <will@bootbonnet.ca>

teoli2003 committed:
Remove note that was a duplicating info

teoli2003 committed:
Moved notes out of Value to the general intro

teoli2003 committed:
Fix with numerous suggestions from the code review

Co-authored-by: wbamberg <will@bootbonnet.ca>

teoli2003 committed:
Is a -> A

teoli2003 committed:
Remove extraneous empty line

teoli2003 committed:
Makes the MD link be on one single line

teoli2003 committed:
Remove DOMString and list syntax without []

teoli2003 committed:
Remove DOMString in Event.type

teoli2003 committed:
Fix Event.timestamp

teoli2003 committed:
Add missing space

teoli2003 committed:
Add missing redirects + typo

teoli2003 commented:
Thanks for the review! I think I've addressed the points. (I also found a couple of details and added two redirects as discussed yesterday during the editorial meeting).

teoli2003 committed:
Remove "For a list of available event types…" paragraph

wbamberg reviewed (approved):
👍 looks great, thank you!
